libs/design/tree/src/tree-item/tree-item.component.spec.ts

```diff
@@ -0,0 +1,333 @@
+      fixture = TestBed.createComponent(UsageWrapperComponent);
+      wrapper = fixture.componentInstance;
+
+      fixture.detectChanges();
+
+      daffTreeItem = fixture.debugElement.query(By.css('daff-tree-item')).componentInstance;
+      treeHeader = fixture.debugElement.query(By.css('.daff-tree-item__header'));
+    });
+
+    it('should create', () => {
+      expect(daffTreeItem).toBeTruthy();
+    });
+
+    it('should set _open to false by default', () => {
+      expect(daffTreeItem._open).toEqual(false);
+    });
+
+    it('should set _animationState to void by default', () => {
+      expect(daffTreeItem._animationState).toEqual('void');
+    });
+
+    it('should be able to accept an initiallyOpen input', () => {
+      wrapper.initiallyOpenValue = false;
+
+      fixture.detectChanges();
+
+      expect(daffTreeItem.initiallyOpen).toEqual(false);
+
+      wrapper.initiallyOpenValue = true;
+
+      fixture.detectChanges();
+
+      expect(daffTreeItem.initiallyOpen).toEqual(true);
+    });
+
+    describe('ngOnInit', () => {
+
+      describe('when initiallyOpen is true', () => {
+
+        beforeEach(() => {
+          wrapper.initiallyOpenValue = true;
+          fixture.detectChanges();
+        });
+
+        it('should set _open to true', () => {
+          daffTreeItem.ngOnInit();
+          expect(daffTreeItem._open).toBeTruthy();
+        });
+      });
+
+      describe('when initiallyOpen is not set', () => {
+
+        beforeEach(() => {
+          wrapper.initiallyOpenValue = undefined;
+          fixture.detectChanges();
+        });
+
+        it('should set open to false', () => {
+          daffTreeItem.ngOnInit();
+          expect(daffTreeItem._open).toBeFalsy();
+        });
+      });
+    });
+
+    describe('when tree header is clicked', () => {
+
+      beforeEach(() => {
+        spyOn(daffTreeItem, 'toggleOpen');
+
+        treeHeader.nativeElement.click();
+      });
+
+      it('should call toggleOpen', () => {
+        expect(daffTreeItem.toggleOpen).toHaveBeenCalledWith();
+      });
+    });
+
+    describe('toggleOpen', () => {
+      it('should toggle open', () => {
+        daffTreeItem._open = false;
+
+        daffTreeItem.toggleOpen();
+        expect(daffTreeItem._open).toEqual(true);
+
+        daffTreeItem.toggleOpen();
+        expect(daffTreeItem._open).toEqual(false);
+      });
+
+      it('should toggle _animationState between void and open', () =>  {
+        daffTreeItem.toggleOpen();
+        expect(daffTreeItem._animationState).toEqual('open');
+
+        daffTreeItem.toggleOpen();
+        expect(daffTreeItem._animationState).toEqual('void');
+      });
+    });
+  });
+
+	@Component({ template: `
+	<daff-tree>
+		<daff-tree-item>
+			<daff-tree>
+				<daff-tree-item [initiallyOpen]="initiallyOpenValue">
+					<h3 daffTreeItemTitle>Size and Fit</h3>
+					<div daffTreeItemContent>no content</div>
+				</daff-tree-item>
+			</daff-tree>
+		</daff-tree-item>
+	</daff-tree>
```

Comment:
Is it necessary to nest the `daff-tree`s for this test? Seems confusing to figure out what you mean by "the first child tree-item" when they're nested like this.

Reply:
I didn't write the test, but isn't it testing that nested trees should have levels added to match the depth of nested tree items?

Would this make more sense: `when it is the parent tree-item of nested trees`?